iroha/src/smartcontracts.rs

```diff
@@ -0,0 +1,52 @@
+//! Module that opts different smartcontract runtime
+//!
+//! Currently supported only Iroha instructions
+
+pub mod isi;
+
+use std::error::Error;
+
+use iroha_data_model::prelude::*;
+pub use isi::*;
+
+use super::wsv::WorldStateView;
+
+/// Trait implementations should provide actions to apply changes on `WorldStateView`.
+#[allow(clippy::missing_errors_doc)]
+pub trait Execute {
+    /// Error type returned by execute function
+    type Error: Error;
```

Comment:
It is better to introduce it in a separate PR. Although I don't strictly see a need right now. Also if we introduce it for Execute then we also need it for Evaluate.

Reply:
The point is that I wanted to make it in another module, while error type will be ISI local.

Reply:
What about adding this also to Evaluate?

Reply:
In future yes. But right now it just has iroha_error::Error everywhere.

Reply:
The same in execute - it is just iroha_error::Error everywhere

Reply:
Actually no. It is typed enough in execute